### PR TITLE
MINOR: [PARQUET] Clean up Skip method naming rows->values

### PR DESCRIPTION
**Pull request by fatemehp**
The Skip method is skipping values and not rows. "values" is used throughout the code interchangeably with levels. Repeated fields may have multiple values, thus the use of "rows" is not accurate because we are not skipping over the values from the repeated field to the next row.

Similarly, two other variables total_num_rows_ and seen_num_rows_ actually refer to values and not rows. So I updated them as well.

I will add more tests for the Skip method that will clarify this behavior for repeated fields in a separate change.

### CODE REVIEW

**Comment by emkornfield** [comment]
LGTM, I'll merge in a few hours unless anybody has more feedback.

**Comment by emkornfield** [comment]
Actually it looks like the lint error is related to this change, lets fix that first

File: cpp/src/parquet/column_reader.h [hunk 1]
```diff
@@ -218,9 +218,11 @@ class TypedColumnReader : public ColumnReader {
                                   int64_t valid_bits_offset, int64_t* levels_read,
                                   int64_t* values_read, int64_t* null_count) = 0;
 
-  // Skip reading levels
-  // Returns the number of levels skipped
-  virtual int64_t Skip(int64_t num_rows_to_skip) = 0;
+  // Skip reading values.
+  // Returns the number of values skipped.
+  // This function will NOT skip rows, and repeated fields may have multiple values
+  // corresponding to the same row.
```
Comment (pitrou) [inline cpp/src/parquet/column_reader.h:224]: I find this comment confusing. Let's say I ask to skip 20 values:
* If I am in a non-repeated field, nothing is skipped and 0 is returned?
* If I am in a repeated field and the current row has 10 values, 10 is returned? How do I skip several rows?

Comment (fatemehp) [reply cpp/src/parquet/column_reader.h]: I clarified the comment, please take a look and see if it makes it clear. Thanks!

File: cpp/src/parquet/column_reader.h [hunk 2]
```diff
@@ -218,10 +218,13 @@ class TypedColumnReader : public ColumnReader {
                                   int64_t valid_bits_offset, int64_t* levels_read,
                                   int64_t* values_read, int64_t* null_count) = 0;
 
-  // Skip reading values.
+  // Skip reading values. This method will work for both repeated and
+  // non-repeated fields. Note that this method is skipping values and not
+  // records. This distinction is important for repeated fields, meaning that
+  // we are not skipping over the values to the next record. We are skipping
+  // through them. So after the skip the iterator could be in the middle of a
```
Comment (pitrou) [inline cpp/src/parquet/column_reader.h:226]: Thanks for the update. This reads better, but I'm afraid the distinction between "skipping over" and "skipping through" is not clear to me. Could you perhaps try to explain better?

Comment (fatemehp) [reply cpp/src/parquet/column_reader.h]: I further clarified the comment with an example, please take a look.

**Review by emkornfield — APPROVED** [review]

**Review by pitrou — APPROVED** [review]
Thanks for the update!

**Comment by emkornfield** [comment]
@jonkeane @thisisnic it looks like the R failure are unrelated to this PR, but wanted to double check before merging.

**Comment by thisisnic** [comment]
@emkornfield Correct, that one is are unrelated.

**Comment by pitrou** [comment]
Thank you for contributing this @fatemehp. PR merged now.

**Pull request merged**